### PR TITLE
Add onnx support

### PR DESCRIPTION
**Pull request by maheshambule**
- Adds ONNX support. This is WIP.
- Tested save, apply and serve APIs for basic sklearn models
- TODOs are listed in code

### CODE REVIEW

File: tests/contrib/test_onnx.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    model_type.load(artifacts)
+    np.testing.assert_array_almost_equal(
+        ONNXWrappedModel(model).predict(inp_data_nparray)[0],
+        model_type.call_method("predict", inp_data_nparray)[0],
+    )
+    assert set(model_type.get_requirements().modules) == expected_requirements
+
+
+# Copyright 2019 Zyfra
```
Comment (mike0sv) [inline tests/contrib/test_onnx.py:148]: This is not needed :)

**Review by mike0sv — CHANGES_REQUESTED** [review]
Had some questions and suggestions, but looks great overall.

File: mlem/contrib/onnx.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        for model_input, input_data in zip(self.model.graph.input, data):
+            input_dict[model_input.name] = convert_to_numpy(
+                input_data, get_onnx_to_numpy_type(model_input)
+            )
+
+        label_names = [out.name for out in self.runtime_session.get_outputs()]
+        pred_onnx = self.runtime_session.run(label_names, input_dict)
+
+        output = []
+        for input_data in pred_onnx:
```
Comment (omesser) [inline mlem/contrib/onnx.py:113]:
```suggestion
        for output_data in pred_onnx:
```
😉

**Review by omesser — CHANGES_REQUESTED** [review]

**Review by omesser — APPROVED** [review]

File: mlem/contrib/onnx.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+from functools import cached_property
```
Comment (mike0sv) [inline mlem/contrib/onnx.py:1]: cached_property is not available in python 3.7

**Review by mike0sv — COMMENTED** [review]
Can you resolve conflicts and make it pass all the checks?

**Review by aguschin — CHANGES_REQUESTED** [review]

**Comment by aguschin** [comment]
@maheshambule, most of the tests pass 🎉, but looks like there are couple of easy tests that needs fixing. Could you please TAL?

**Review by aguschin — APPROVED** [review]

**Comment by aguschin** [comment]
Yes, it was enough to retry the test. We need to mark some tests flaky to retry them instead of failing CI, #326 
Great work on supporting onnx btw! @mike0sv, do you want to TAL or I can go ahead and merge this?

**Comment by aguschin** [comment]
Sorry @maheshambule, missed your last message. Just triggered the CI :)

**Review by aguschin — APPROVED** [review]

**Review by omesser — APPROVED** [review]

File: tests/contrib/test_pandas.py [hunk 4]
```diff
@@ -617,7 +617,12 @@ def f(x):
 
     sig = Signature.from_method(f, auto_infer=True, x=data)
 
-    assert get_object_requirements(sig).modules == ["pandas", "numpy"]
+    actual, expected = get_object_requirements(sig).modules, [
+        "pandas",
+        "numpy",
+    ]
+    assert len(actual) == len(expected)
+    assert all([a == b for a, b in zip(actual, expected)])
```
Comment (aguschin) [inline tests/contrib/test_pandas.py:625]:
```suggestion
    assert all(a == b for (a, b) in zip(actual, expected))
```

Comment (aguschin) [reply tests/contrib/test_pandas.py:625]: to satisfy pylint

**Pull request merged**